Explorer/Assets/DCL/AvatarRendering/Emotes/Systems/LoadSceneEmotesSystem.cs

```diff
@@ -0,0 +1,167 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using CommunicationData.URLHelpers;
+using DCL.AvatarRendering.Wearables;
+using DCL.Diagnostics;
+using ECS.Abstract;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.AssetBundles;
+using ECS.StreamableLoading.Common.Components;
+using System;
+using UnityEngine;
+using Utility;
+using StreamableResult = ECS.StreamableLoading.Common.Components.StreamableLoadingResult<DCL.AvatarRendering.Emotes.EmotesResolution>;
+using AssetBundlePromise = ECS.StreamableLoading.Common.AssetPromise<ECS.StreamableLoading.AssetBundles.AssetBundleData, ECS.StreamableLoading.AssetBundles.GetAssetBundleIntention>;
+
+namespace DCL.AvatarRendering.Emotes
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [LogCategory(ReportCategory.EMOTE)]
+    public partial class LoadSceneEmotesSystem : BaseUnityLoopSystem
+    {
+        private const string SCENE_EMOTE_PREFIX = "urn:decentraland:off-chain:scene-emote";
+
+        private readonly URLSubdirectory customStreamingSubdirectory;
+        private readonly IEmoteCache emoteCache;
+
+        public LoadSceneEmotesSystem(World world,
+            IEmoteCache emoteCache,
+            URLSubdirectory customStreamingSubdirectory)
+            : base(world)
+        {
+            this.emoteCache = emoteCache;
+            this.customStreamingSubdirectory = customStreamingSubdirectory;
+        }
+
+        protected override void Update(float t)
+        {
+            GetEmotesFromRealmQuery(World, t);
+        }
+
+        [Query]
+        [None(typeof(StreamableResult))]
+        private void GetEmotesFromRealm([Data] float dt, in Entity entity,
+            ref GetSceneEmoteFromRealmIntention intention,
+            ref IPartitionComponent partitionComponent)
+        {
+            if (intention.CancellationTokenSource.IsCancellationRequested)
+            {
+                if (!World.Has<StreamableResult>(entity))
+                    World.Add(entity, new StreamableResult(new OperationCanceledException($"Scene emote request cancelled {intention.Hash}")));
+
+                return;
+            }
+
+            intention.ElapsedTime += dt;
+
+            URN urn = GetUrn(intention.Hash, intention.Loop);
+
+            bool isTimeout = intention.ElapsedTime >= intention.Timeout;
+
+            if (isTimeout)
+            {
+                if (!World.Has<StreamableResult>(entity))
+                {
+                    ReportHub.LogWarning(GetReportCategory(), $"Loading scenes emotes timed out {urn}");
+                    World.Add(entity, new StreamableResult(new TimeoutException($"Scene emote timeout {urn}")));
+                }
+
+                return;
+            }
+
+            if (!emoteCache.TryGetEmote(urn, out IEmote emote))
+            {
+                if (!intention.IsModelProcessed)
```

Comment:
Can you add an empty dummy related to the urn? If not `intention.IsModelProcessed` looks redundant: the process of adding a DTO is fully synchronous and corresponds to the presence in the cache